src/librustc/ty/subst.rs

```diff
@@ -384,9 +409,9 @@ impl<'tcx> TypeFoldable<'tcx> for SubstsRef<'tcx> {
         // If folding doesn't change the substs, it's faster to avoid
         // calling `mk_substs` and instead reuse the existing substs.
         if params[..] == self[..] {
-            self
+            *self
         } else {
-            folder.tcx().intern_substs(&params)
+            folder.tcx().intern_substs(&params).into()
```

Comment:
`intern_substs` should directly return a `SubstsRef`

Reply:
@oli-obk 
As `_intern_substs` comes from [slice_interners!](https://github.com/rust-lang/rust/blob/d4a32d504a5aa49b951bfc70602a9615cb772acf/src/librustc/ty/context.rs#L2375-L2393), which returns a `List`, then should I implement a specific one for `SubstsRef`?

Reply:
You should be able to just modify that method, too. I don't think it's used elsewhere.